new-lamassu-admin/src/pages/Dashboard/SystemPerformance/SystemPerformance.js

```diff
@@ -0,0 +1,293 @@
+                getDateSecondsAgo(30 * 24 * 3600, now)
+              )
+            )
+          }
+          return (
+            t.error === null &&
+            moment(t.created).isBetween(
+              getDateSecondsAgo(30 * 24 * 3600, now),
+              now
+            )
+          )
+        default:
+          return t.error === null && true
+      }
+    }
+
+    const convertFiatToLocale = item => {
+      if (item.fiatCode === fiatLocale) return item
+      const itemRate = R.find(R.propEq('code', item.fiatCode))(data.btcRates)
+      const localeRate = R.find(R.propEq('code', fiatLocale))(data.btcRates)
+      const multiplier = localeRate.rate / itemRate.rate
+      return { ...item, fiat: parseFloat(item.fiat) * multiplier }
+    }
+
+    setTransactionsToShow(
+      R.map(convertFiatToLocale)(
+        R.filter(isInRange(false), data?.transactions ?? [])
+      )
+    )
+    setTransactionsLastTimePeriod(
+      R.map(convertFiatToLocale)(
+        R.filter(isInRange(true), data?.transactions ?? [])
+      )
+    )
+  }, [data, fiatLocale, selectedRange])
+
+  const handleSetRange = range => {
+    setSelectedRange(range)
+  }
+
+  const getNumTransactions = () => {
+    return R.length(R.filter(isNotProp('error'), transactionsToShow))
+  }
+
+  const getFiatVolume = () => {
+    // for explanation check https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/round
+    return +(
+      Math.round(
+        R.sum(getFiats(R.filter(isNotProp('error'), transactionsToShow))) +
+          'e+2'
+      ) + 'e-2'
+    )
+  }
+
+  const getProfit = (transactions = transactionsToShow) => {
+    const cashInFees = R.sum(
+      getProps('cashInFee')(R.filter(isNotProp('error'), transactions))
+    )
+    let commissionFees = 0
+    transactions.forEach(t => {
+      if (t.error === null) {
+        commissionFees +=
+          Number.parseFloat(t.commissionPercentage) * Number.parseFloat(t.fiat)
+      }
+    })
+    return +(Math.round(commissionFees + cashInFees + 'e+2') + 'e-2')
+  }
+
+  const getPercentChange = () => {
+    const thisTimePeriodProfit = getProfit(transactionsToShow)
+    const previousTimePeriodProfit = getProfit(transactionsLastTimePeriod)
+    if (previousTimePeriodProfit === 0) {
+      return 100
+    }
+    return Math.round(
+      (100 * (thisTimePeriodProfit - previousTimePeriodProfit)) /
+        Math.abs(previousTimePeriodProfit)
+    )
+  }
+
+  const getDirectionPercent = () => {
+    const directions = {
+      cashIn: 0,
+      cashOut: 0,
+      length: 0
+    }
+    transactionsToShow.forEach(t => {
+      if (t.error === null) {
+        switch (t.txClass) {
+          case 'cashIn':
+            directions.cashIn += 1
+            directions.length += 1
+            break
+          case 'cashOut':
+            directions.cashOut += 1
+            directions.length += 1
+            break
+          default:
+            break
+        }
+      }
+    })
```

Comment:
Filter the tx by errors, partition into two arrays (cashin, cashout).

Then you can get length for each array, and total length is just a sum of the two.